python/lsst/jointcal/jointcal.py

```diff
@@ -885,15 +885,30 @@ def _load_data(self, inputSourceTableVisit, inputVisitSummary, associations,
             catalogMap = {ref.dataId['visit']: i for i, ref in enumerate(inputSourceTableVisit)}
             detectorDict = {detector.getId(): detector for detector in camera}
 
+            columns = None
+
             for visitSummaryRef in inputVisitSummary:
                 visitSummary = visitSummaryRef.get()
-                visitCatalog = inputSourceTableVisit[catalogMap[visitSummaryRef.dataId['visit']]].get()
+
+                dataRef = inputSourceTableVisit[catalogMap[visitSummaryRef.dataId['visit']]]
+                if isinstance(dataRef, dafPersist.ButlerDataRef):
```

Comment:
This shouldn't be necessary here at all: this method is only called when running in gen3.